src/main/java/com/um/inventoryservice/BusinessLayer/EmployeeServiceImpl.java

```diff
@@ -27,4 +27,22 @@ public Mono<EmployeeDTO> insertEmployee(Mono<EmployeeDTO> employeeDTOMono) {
                 .flatMap((employeeRepository::save))
                 .map(EntityDTOUtil::toDTO);
     }
+
+    @Override
+    public Mono<EmployeeDTO> getEmployeeById(String employeeId) {
+        return employeeRepository.findEmployeeByEmployeeId(employeeId)
+                .map(EntityDTOUtil::toDTO);
+    }
+
+    @Override
+    public Mono<EmployeeDTO> updateEmployee(String employeeId, Mono<EmployeeDTO> employeeDTOMono) {
+        return employeeRepository.findEmployeeByEmployeeId(employeeId)
+                .flatMap(e -> employeeDTOMono
+                        .map(EntityDTOUtil::toEntity)
+                        .doOnNext(x -> x.setEmployeeId(e.getEmployeeId()))
+                        .doOnNext(x -> x.setId(e.getId()))
```

Comment:
do we need to set both the employeeId and db id? doesn't it automatically do it once it's added to our db?

Reply:
The "update client" method was used as reference, after review of the employee model and the method, the Id does not need to be set. Line 43 will be removed.